docs/streams/upgrade-guide.html

```diff
@@ -141,6 +141,91 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
 
+    <h4>Early Access of the Streams Rebalance Protocol</h4>
+
+    <p>
+        The Streams Rebalance Protocol is a broker-driven rebalancing system designed specifically for Kafka
+        Streams applications. Following the pattern of KIP-848, which moved rebalance coordination of plain consumers
+        from clients to brokers, KIP-1071 extends this model to Kafka Streams workloads. Instead of clients
+        computing new assignments on the client during rebalance events involving all members of the group, assignments are
+        computed continuously on the broker. Instead of using a consumer group, the streams application registers as a
+        streams group with the broker, which manages and exposes all metadata required for coordination of the
+        streams application instances.
+    </p>
+
+    <p>
+        This Early Access release covers a subset of the functionality detailed in
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-1071%3A+Streams+Rebalance+Protocol">KIP-1071</a>.
+        Do not use the new protocol in production. The API is subject to change in future
+        releases.
+    </p>
+
+    <p><strong>What's Included in Early Access</strong></p>
+
+    <ul>
+        <li><strong>Core Streams Group Rebalance Protocol:</strong> The <code>group.protocol=streams</code> configuration
+            enables the dedicated streams rebalance protocol. This separates streams groups from consumer groups and
+            provides a streams-specific group membership lifecycle and metadata management on the broker.</li>
+        <li><strong>Sticky Task Assignor:</strong> A basic task assignment strategy that minimizes task movement
+            during rebalances is included.</li>
+        <li><strong>Interactive Query Support:</strong> IQ operations are compatible with the new streams protocol.</li>
+        <li><strong>New Admin RPC:</strong> The <code>StreamsGroupDescribe</code> RPC provides streams-specific metadata
+            separate from consumer group information, with corresponding access via the <code>Admin</code> client.</li>
+        <li><strong>CLI Integration:</strong> You can list, describe, and delete streams groups via the <code>kafka-streams-groups.sh</code> script.</li>
+    </ul>
+
+    <p><strong>What's Not Included in Early Access</strong></p>
+
+    <ul>
+        <li><strong>Static Membership:</strong> Setting a client `instance.id` will be rejected.</li>
+        <li><strong>Topology Updates:</strong> If a topology is changed significantly (e.g., by adding new source topics
+            or changing the number of sub-topologies), a new streams group must be created.</li>
+        <li><strong>High Availability Assignor:</strong> Only the sticky assignor is supported.</li>
+        <li><strong>Regular Expressions:</strong> Pattern-based topic subscription is not supported.</li>
+        <li><strong>Reset Operations:</strong> CLI offset reset operations are not supported.</li>
```

Comment:
This might to get updated, in case we get this feature into 4.1. (Just comment as a reminder in case we cherry-pick this PR).

Reply:
Not included